### PR TITLE
Cleanup pass on Transaction abstractions and public surface.

### PR DESCRIPTION
**Pull request by jason-bragg**
- Removed leftover code from singleton TM.
- Moves agent abtraction from core to runtime abstractions
- Cleaned up agent initialization and usage pattern
- Reduced public surface of transactional state abstraction
- Moved transaction info out of abstractions
- Removed old unsused logging errorcodes.
- Moved serialization generation assembly statements to files that defined the serializable classes.

### CODE REVIEW

File: src/Orleans.Transactions/ErrorCodes.cs [hunk 1]
```diff
@@ -1,61 +1,15 @@
...
 {
     /// <summary>
-    /// Orleans ServiceBus error codes
+    /// Orleans Transactions error codes
     /// </summary>
     internal enum OrleansTransactionsErrorCode
     {
         /// <summary>
-        /// Start of orlean servicebus errocodes
+        /// Start of orlean transactions errocodes
```
Comment (ReubenBond) [inline src/Orleans.Transactions/ErrorCodes.cs]: Orleans
error codes

File: src/Orleans.Transactions/ErrorCodes.cs [hunk 2]
```diff
@@ -1,61 +1,15 @@
...
-        internal static void Warn(this ILogger logger, OrleansTransactionsErrorCode errorCode, string message, Exception exception)
-        {
-            logger.Warn((int)errorCode,  message, new object[] { }, exception);
-        }
-
-        internal static void Error(this ILogger logger, OrleansTransactionsErrorCode errorCode, string message, Exception exception = null)
-        {
-            logger.Error((int)errorCode, message, exception);
-        }
+        // TODO - jbragg - add error codes for transaction errors - User Story 62
```
Comment (ReubenBond) [inline src/Orleans.Transactions/ErrorCodes.cs]: IMO we should add a link to the story in VSO - it's easy to forget where User Story 62 lives

File: src/Orleans.Transactions/Abstractions/ITransactionalStateStorage.cs [hunk 3]
```diff
@@ -1,7 +1,11 @@
-﻿using Orleans.Concurrency;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using Orleans.CodeGeneration;
+using Orleans.Concurrency;
+
+[assembly: GenerateSerializer(typeof(Orleans.Transactions.Abstractions.PendingTransactionState<>))]
+[assembly: GenerateSerializer(typeof(Orleans.Transactions.Abstractions.TransactionalStorageLoadResponse<>))]
```
Comment (ReubenBond) [inline src/Orleans.Transactions/Abstractions/ITransactionalStateStorage.cs]: I prefer assembly-level attributes live in a file called AssemblyInfo.cs which only contains assembly-level metadata

File: src/Orleans.Transactions/ErrorCodes.cs (hunk 2 above)
Comment (jason-bragg) [reply src/Orleans.Transactions/ErrorCodes.cs]: Tasking is a mess right now, will remove reference to specific task as odds of link or task remaining the same by the time we get to fixing this is fairly weak.

**Pull request merged**